LibZipSharp/Xamarin.Tools.Zip/ZipArchive.cs

```diff
@@ -809,8 +858,18 @@ internal static unsafe Int64 stream_callback (IntPtr state, IntPtr data, UInt64
 					} finally {
 						ArrayPool<byte>.Shared.Return (buffer);
 					}
-
 				case SourceCommand.BeginWrite:
+					try {
+						string tempFile = Path.GetTempFileName ();
+						context.Destination = File.Open (tempFile, FileMode.OpenOrCreate, FileAccess.ReadWrite);
+						context.DestinationFileName = tempFile;
+					} catch (IOException) {
+						// ok use a memory stream as a backup
```

Comment:
@grendello do you think this is a good idea? or should we just throw an exception and fail.

Reply:
I think it's fine, we should do our best to make the code work